### ISSUE TITLE
MA0015 should not be considered an equivalent of CA2208

### ISSUE THREAD

**Issue by drieseng** [meziantou/Meziantou.Analyzer#464]
**CA2208** does not report a diagnostic if you use an overload of the **ArgumentException** ctor that does not take a parameter name.
```c#
void Sample1(string str)
{
    // MA0015: Use an overload of 'System.ArgumentException' with the parameter name
    // CA2208: Not reported!
    throw new ArgumentException("Message");
}

void Sample2(string str)
{
    // MA0015: Use an overload of 'System.ArgumentException' with the parameter name
    // CA2208: Not reported!
   throw new ArgumentException("Message", new InvalidOperationException());
}
```

**CA2208** only reports diagnostics for **ArgumentException**, not for derived types:

```c#
void Sample1(string str)
{
    // MA0015: 'x' is not a valid parameter name
    // CA2208: Not reported!
    throw new ArgumentNullException("x", "Message");
}

void Sample2(string str)
{
    // MA0015: Use an overload of 'System.ArgumentNullException' with the parameter name
    // CA2208: Not reported!
    throw new ArgumentNullException("Message", new InvalidOperationException());
}
```

I propose to:
* move the **CA2208** / **MA0015** combination from the **Equivalent rules** to the **Similar rules** section of the **Comparison with other analyzers** document.
* document that **MA0015** also applies to types that derive from **ArgumentException**.

**Comment by meziantou** [comment]
Looks good to me! Feel free to submit a PR to update the documentation.

**Comment by drieseng** [comment]
I'll do that later this week.

**Comment by drieseng** [comment]
I submitted https://github.com/dotnet/roslyn-analyzers/issues/6580 for the issue in **CA2208**  that I mentioned above.
